x-pack/plugins/infra/public/pages/metrics/inventory_view/hooks/use_timeline.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { fold } from 'fp-ts/lib/Either';
+import { identity } from 'fp-ts/lib/function';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { first } from 'lodash';
+import { useEffect, useMemo, useCallback } from 'react';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { getIntervalInSeconds } from '../../../../../server/utils/get_interval_in_seconds';
+import { throwErrors, createPlainError } from '../../../../../common/runtime_types';
+import { useHTTPRequest } from '../../../../hooks/use_http_request';
+import {
+  SnapshotNodeResponseRT,
+  SnapshotNodeResponse,
+  SnapshotRequest,
+  InfraTimerangeInput,
+} from '../../../../../common/http_api/snapshot_api';
+import {
+  InventoryItemType,
+  SnapshotMetricType,
+} from '../../../../../common/inventory_models/types';
+
+const ONE_MINUTE = 60;
+const ONE_HOUR = ONE_MINUTE * 60;
+const ONE_DAY = ONE_HOUR * 24;
+const ONE_WEEK = ONE_DAY * 7;
+
+const getTimeLengthFromInterval = (interval: string | undefined) => {
+  if (interval) {
+    const intervalInSeconds = getIntervalInSeconds(interval);
+    const multiplier =
+      intervalInSeconds < ONE_MINUTE
+        ? ONE_HOUR / intervalInSeconds
+        : intervalInSeconds < ONE_HOUR
+        ? 60
+        : intervalInSeconds < ONE_DAY
+        ? 7
+        : intervalInSeconds < ONE_WEEK
+        ? 30
+        : 1;
+    const timeLength = intervalInSeconds * multiplier;
+    return { timeLength, intervalInSeconds, lookbackSize: Math.max(6, multiplier) };
+  } else {
+    return { timeLength: 0, lookbackSize: 0, intervalInSeconds: 0 };
+  }
+};
+
+export function useTimeline(
+  filterQuery: string | null | undefined,
+  metrics: Array<{ type: SnapshotMetricType }>,
+  nodeType: InventoryItemType,
+  sourceId: string,
+  currentTime: number,
+  accountId: string,
+  region: string,
+  interval: string | undefined,
+  shouldReload: boolean
+) {
+  const decodeResponse = (response: any) => {
+    return pipe(
+      SnapshotNodeResponseRT.decode(response),
+      fold(throwErrors(createPlainError), identity)
+    );
+  };
+
+  const timeLengthResult = useMemo(() => getTimeLengthFromInterval(interval), [interval]);
+  const { timeLength, lookbackSize, intervalInSeconds } = timeLengthResult;
+
+  const timerange: InfraTimerangeInput = {
```

Comment:
Are you trying to workaround the lookback limitation? If you just set the `from`, `to`, `interval` and then set `ignoreLookback: true` it would draw all the buckets without having to calculate the lookback.

Reply:
Thanks, fixing that